apps/user_status/lib/ResponseDefinitions.php

```diff
@@ -22,7 +22,7 @@
  *     icon: string,
  *     message: string,
  *     clearAt: ?UserStatusClearAt,
- *     visible: ?bool,
+ *     visible?: bool,
```

Comment:
I want to check this on Monday, so please don't merge before

Reply:
The only place returning/exposing `UserStatusPredefined` is filtering by that attribute, so it is actually:
```suggestion
 *     visible: true,
```

But it sounds like the field should never have been exposed at all, as it's internal preparation. So I would say the best option would be:
```suggestion
```

But that needs some code in the controller to convince Psalm

Reply:
> But that needs some code in the controller to convince Psalm

A simple `unset()` should do it

Reply:
yes, just meant it needs to touch the actual code

Reply:
Hopefully no client would use this field yet ;)

Reply:
Right, because this is in fact a breaking change.
So either we go with my change or we always set it to true (which has the same meaning as undefined here).

Reply:
But your change widens the API and could subsequently break it :P 
I'd say we change it to:
```suggestion
 *     visible: true,
```
Because that is what is currently returned on the API

Reply:
The relevant code is `return !array_key_exists('visible', $status) || $status['visible'] === true;` which means the value can be either `true` or not present. So the more correct signature would be `visible?: true`, but it has to be marked as not required either way.

Reply:
Ah but I think we can actually safely remove the field from the output. It is either not present or `false` in the data, so it actually never appears in the output. Even though the spec was broken before, any client must have already handled this in some way.